BMS Workspace-git/bms-1227-fw/Phantom_drivers/rtos/source/task_command_line.c

```diff
@@ -0,0 +1,43 @@
+/*
+ * task_command_line.c
+ *
+ *  Created on: Oct 25, 2022
+ *      Author: lonam
+ */
+#include "task_command_line.h"
+
+#include "FreeRTOS.h"
+#include "FreeRTOSConfig.h"
+#include "os_task.h"
+
+#include "Phantom_sci.h"
+#include "hwConfig.h"
+
+static uint8 previous;
+static xTaskHandle commandLineTaskHandle;
+static void CommandLineTask(void* params){
+    while(1){
+        uint8 msg;
+        sciReceive(PC_UART, 1, &msg);
+        if(msg==previous){
+            // do nothing
+        } else if(msg=='1'){
```

Comment:
when creating actual commands let's use proper macros. This makes it easier to read and change and uses no program memory because the values are placed at compile time

```
#define DO_SMT_CMD '1'

if (msg == DO_SMT_CMD)
...
```